src/test/java/com/amazonaws/services/sqs/util/IntegrationTest.java

```diff
@@ -40,4 +61,44 @@ public void teardownSQSClient() {
         }
         exceptionHandler.assertNothingThrown();
     }
+
+    protected AmazonSQS getBuddyPrincipalClient() {
+        AWSCredentialsProvider credentialsProvider = new ProfileCredentialsProvider("buddy");
+        try {
+            credentialsProvider.getCredentials();
+        } catch (Exception e) {
+            assumeNoException("This test requires a second 'buddy' credential profile.", e);
+        }
+
+        AmazonSQS client = AmazonSQSClientBuilder.standard()
+                .withRegion("us-west-2")
+                .withCredentials(credentialsProvider)
+                .build();
+
+        // Assume that the principal is not able to send messages to arbitrary queues
+        String queueUrl = sqs.createQueue(queueNamePrefix + "TestQueue").getQueueUrl();
+        try {
+            client.sendMessage(queueUrl, "Haxxors!!");
+            assumeTrue("The buddy credentials should not authorize sending to arbitrary queues", false);
+        } catch (AmazonSQSException e) {
+            // Access Denied
+            assumeThat(e.getStatusCode(), equalTo(403));
+        } finally {
+            sqs.deleteQueue(queueUrl);
+        }
+
+        return client;
+    }
+
+    protected Policy allowSendMessagePolicy() {
+        Policy policy = new Policy();
+        Statement statement = new Statement(Statement.Effect.Allow);
+        statement.setActions(Collections.singletonList(SQSActions.SendMessage));
+        // Ideally we would only allow the principal we're testing with, but we
```

Comment:
It could be retrieve through `sts get-caller-identity`, which does not requires any permission.